R/util.R

```diff
@@ -514,42 +514,65 @@ util.assign_list_to_environment <- function(l, environment = .GlobalEnv){
     }
 }
 
-util.list_files <- function (initial_path, max_depth = 2, current_depth = 0) {
+util.list_all <- function (initial_path, max_depth = 2, type = 'all', current_depth = 0) {
     # Like base::list.files, but better:
-    # * Lists files, not directories. Because it's called list_files.
+    # * Can choose to list only files, not directories (type = 'files').
     # * Rather than choosing between no recursion (not very useful) and full
     #   recursion (potentially very slow for a deep folder tree) you can set a
     #   max depth. Choose 0 for no recursion.
     # * Ignores hidden and system files, defined as anything starting with
     #   '.', '$', or '~'.
-    # * Only works on unix-like systems (no windows)!
     #
     # Args:
     #   initial_path: atomic char, directory to scan for files.
     #   max_depth: atomic int, default 2, how many subfolders deep to scan for
     #     files. Zero means enter no subfolders.
-    # current_depth: internal use only, do not specify.
+    #   type: atomic char, default 'all', or choose 'dirs' or 'files'.
+    #   current_depth: internal use only, do not specify.
```

Comment:
That's how I designed the recursion. Each time it recurses, it passes the current depth to the next call; that way it knows when to stop. You're not supposed to set it b/c this function has two modes: a human calling it, and it calling itself. The parameter is only used in the second case.